File: build-array-from-permutation/Python3/build-array-from-permutation.py
class Solution:
    def buildArray(self, nums: list[int]) -> list[int]:
        return [nums[nums[i]] for i in range(len(nums))]


input = [[0, 2, 1, 5, 3, 4], [5, 0, 1, 2, 3, 4]]
output = [[0, 1, 2, 4, 5, 3], [4, 5, 0, 1, 2, 3]]
for i in range(len(input)):
    r = Solution().buildArray(input[i])
    if r != output[i]:
        raise Exception('Failed: ' + str(input[i]) + ' ---- Got: ' + str(r) + ' !== ' + str(output[i]))
    print('Passed input: ' + str(input[i]))

File: build-array-from-permutation/JavaScript/build-array-from-permutation.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
const buildArray = function (nums) {
    return nums.map(n=>nums[n]);
};

const inputs = [[0,2,1,5,3,4], [5,0,1,2,3,4]]
const outputs = [[0,1,2,4,5,3], [4,5,0,1,2,3]]
inputs.forEach((input,i)=>{
  const r = buildArray(input);
  if(JSON.stringify(r)!==JSON.stringify(outputs[i]))
      throw new Error('Failed: '+input+' ---- Got: '+JSON.stringify(r)+' !== '+JSON.stringify(outputs[i]));
  else console.log('Passed input: '+input);
});